Src/Metrics/Reporters/PrometheusReport.cs

```diff
@@ -0,0 +1,188 @@
+﻿using System;
+using System.Text;
+using System.Text.RegularExpressions;
+using Metrics.MetricData;
+using System.Threading;
+
+namespace Metrics.Reporters
+{
+    public sealed class PrometheusReport : BaseReport
+    {
+        private StringBuilder reportText;
+
+        private PrometheusReport()
+        {
+            reportText = new StringBuilder();
+        }
+
+        private static readonly DateTime epoch = new DateTime(1970, 1, 1, 0, 0, 0, DateTimeKind.Utc);
+        private long CurrentTimeMillis()
+        {
+            TimeSpan ts = base.CurrentContextTimestamp - epoch;
+            ts = ts.Duration(); // absolute value, for dates before Jan 1st, 1970
+            long total = ts.Days * 24 + ts.Hours; // total is in hours
+            total *= 60; // in minutes
+            total += ts.Minutes;
+            total *= 60; // in seconds
+            total += ts.Seconds;
+            total *= 1000; // in millis
+            total += ts.Milliseconds;
+            return base.CurrentContextTimestamp >= epoch ? total : -total; // all of this because ts.TotalMilliseconds is a double. *sigh*
```

Comment:
I am worried about floating point inaccuracies, and I just didn't think
about using Ticks. Long story short: brain fart. Fixed.

And what was wrong with ts.TotalMilliseconds was accuracy loss.

On Wed, Jun 1, 2016 at 6:28 PM, Paccc <notifications@github.com> wrote:

> In Src/Metrics/Reporters/PrometheusReport.cs
> <https://github.com/Recognos/Metrics.NET/pull/27#discussion_r65454777>:
>
> > +            reportText = new StringBuilder();
> > +        }
> > +
> > +        private static readonly DateTime epoch = new DateTime(1970, 1, 1, 0, 0, 0, DateTimeKind.Utc);
> > +        private long CurrentTimeMillis()
> > +        {
> > +            TimeSpan ts = base.CurrentContextTimestamp - epoch;
> > +            ts = ts.Duration(); // absolute value, for dates before Jan 1st, 1970
> > +            long total = ts.Days * 24 + ts.Hours; // total is in hours
> > +            total *= 60; // in minutes
> > +            total += ts.Minutes;
> > +            total *= 60; // in seconds
> > +            total += ts.Seconds;
> > +            total *= 1000; // in millis
> > +            total += ts.Milliseconds;
> > +            return base.CurrentContextTimestamp >= epoch ? total : -total; // all of this because ts.TotalMilliseconds is a double. *sigh*
>
> Just out of curiosity, why can't you do something like return ts.Ticks -
> (ts.Ticks % TimeSpan.TicksPerMillisecond);? If you're worried about
> floating point inaccuracies, this way keeps everything as a long without
> having to cast.
> Also, what was wrong with the first way you did it, when you casted
> ts.TotalMilliseconds to a long? Doesn't that result in the same value?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Recognos/Metrics.NET/pull/27/files/9e562b1d08fcbee54a9da7648a5fd0a6d4e12e54..f5c70cdc1b21f2ec13a25345cde2993640ebb6a0#r65454777>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AStJnlNOSG-Mj65K0A6wmoetHOisxtmZks5qHgecgaJpZM4Io0G3>
> .
>



-- 
Peace, Love and Unfinished Signatures,
Christian "Camembert" Lavoie